tests/system/reader/test_reader.py

```diff
@@ -179,7 +180,12 @@ def test_column_selection_read(
 
 def test_snapshot(client_and_types, project_id, table_with_data_ref, bq_client):
     client, types = client_and_types
+    # Sleep for a moment to give use some wiggle room in case the BigQuery
+    # snapshot time and our times are out-of-sync.
+    # https://github.com/googleapis/python-bigquery-storage/issues/151
+    time.sleep(10)
```

Comment:
What is the point of the initial sleep? it seems the second is to ensure that the snapshot time requested is in the past?

Reply:
The flake is actually that there are 0 rows, so the snapshot time is too _early_ in that case.

That said, Seth makes a good point that I can use the times from the table metadata.